src/mon/OSDMonitor.cc

```diff
@@ -3510,15 +3511,33 @@ bool OSDMonitor::preprocess_boot(MonOpRequestRef op)
     const auto &otype = type_iter->second;
     // m->metadata["osd_type"] must be "crimson", classic doesn't send osd_type
     if (otype == "crimson") {
-      if (!osdmap.get_allow_crimson()) {
-	mon.clog->info()
-	  << "Disallowing boot of crimson-osd without allow_crimson "
-	  << "OSDMap flag.  Run ceph osd set-allow-crimson to set "
-	  << "allow_crimson flag.  Note that crimson-osd is "
-	  << "considered unstable and may result in crashes or "
-	  << "data loss.  Its usage should be restricted to "
-	  << "testing and development.";
-	goto ignore;
+      cmdmap_t cmdmap;
+      bool crimson_pools_are_configured = cmd_getval_or<bool>(cmdmap, "crimson", false) ||
+                                        osdmap.is_osd_pool_default_crimson_set(cct);
+      bool all_mandatory_crimson_flags_are_set = osdmap.is_osd_pool_default_crimson_set(cct) &&
+                                  osdmap.get_allow_crimson() &&
+                                  g_ceph_context->check_experimental_feature_enabled("crimson");
+      if (!all_mandatory_crimson_flags_are_set){
+        ostringstream ss, ss2;
```

Comment:
I've left this without a goto reply_no_propose at the end of the block because there's no such thing in this file